packages/workspace/src/schematics/remove/lib/check-imports.ts

```diff
@@ -0,0 +1,75 @@
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { getWorkspace, NxJson, readJsonInTree } from '@nrwl/workspace';
+import { from, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Schema } from '../schema';
+
+/**
+ * Check whether the project to be removed is imported by another project
+ *
+ * Throws an error if the project is in use, unless the `--forceRemove` option is used.
+ *
+ * @param schema The options provided to the schematic
+ */
+export function checkImports(schema: Schema): Rule {
+  if (schema.forceRemove) {
+    return (tree: Tree) => tree;
+  }
+
+  return (tree: Tree, context: SchematicContext): Observable<Tree> => {
+    return from(getWorkspace(tree)).pipe(
+      map(workspace => {
+        const nxJson = readJsonInTree<NxJson>(tree, 'nx.json');
+        const project = workspace.projects.get(schema.projectName);
+
+        if (project.extensions['projectType'] === 'application') {
+          // These shouldn't be imported anywhere
+          return tree;
+        }
+
+        const findImportPath = new RegExp(
+          `@${nxJson.npmScope}/${project.root.substr(5)}`
```

Comment:
Not a huge deal I suppose but this doesn't find projects when a user remaps their tsconfig.json?

Reply:
You mean if someone makes it so the `nx.json` and `tsconfig.json` are out of sync?

I could probably just grab the import path by trying to find it directly in the `tsconfig.json` by using the root folder to match against...?